### ISSUE TITLE
If our video does not initially contain the object we want to detect, how should we modify the code?

### ISSUE THREAD

**Issue by Zhujx-zjjx** [IDEA-Research/Grounded-SAM-2#45]
As stated in the question, when the object I want to detect does not appear at the beginning of my video, the code will report an error when running. What method should I use to eliminate this hidden danger?

**Comment by Zhujx-zjjx** [comment]
I agree with your comments. The problem is that designing a check function might be not easy, could you give me some suggestions about this check function?

**Comment by rentainhe** [comment]
> I agree with your comments. The problem is that designing a check function might be not easy, could you give me some suggestions about this check function?

I think you can try this solution:

- Firstly, run grounding model on a set of frames, like `frame 1`, `frame 3`, `frame 5`, and collect all the grounding results of your specific classname
- Then, you can try to use `NMS` op to filter objects with large overlap areas
- Use the grounding boxes as your initial prompt for tracking

You can try this solution for your scenarios.

And I think this is a good topic for us to discuss how to use SAM 2 better

**Comment by Paul0302** [comment]
I faced this issue recently either, I'm using `grounded_sam2_tracking_demo_custom_gd1.5.py`, and i modify the Grounded-Dino part. First is to check if the `input_boxes` is empty or not, if yes than detect next frame until the object appear. Hope this can help you
